src/workerd/jsg/modules-new.c++

```diff
@@ -706,76 +714,80 @@ v8::MaybeLocal<v8::Module> resolveCallback(v8::Local<v8::Context> context,
   auto& registry = IsolateModuleRegistry::from(isolate);
   auto& js = Lock::from(isolate);
 
-  try {
-    return js.tryCatch([&]() -> v8::MaybeLocal<v8::Module> {
-      auto spec = kj::str(specifier);
-
-      // The proposed specification for import assertions strongly recommends that
-      // embedders reject import attributes and types they do not understand/implement.
-      // This is because import attributes can alter the interpretation of a module.
-      // Throwing an error for things we do not understand is the safest thing to do
-      // for backwards compatibility.
-      //
-      // For now, we do not support any import attributes, so if there are any at all
-      // we will reject the import.
-      if (!import_assertions.IsEmpty()) {
-        if (import_assertions->Length() > 0) {
-          js.throwException(js.typeError("Import attributes are not supported"));
-        };
-      }
+  return js.tryCatch([&]() -> v8::MaybeLocal<v8::Module> {
+    auto spec = kj::str(specifier);
+
+    // The proposed specification for import assertions strongly recommends that
+    // embedders reject import attributes and types they do not understand/implement.
+    // This is because import attributes can alter the interpretation of a module.
+    // Throwing an error for things we do not understand is the safest thing to do
+    // for backwards compatibility.
+    //
+    // For now, we do not support any import attributes, so if there are any at all
+    // we will reject the import.
+    if (!import_assertions.IsEmpty()) {
+      if (import_assertions->Length() > 0) {
```

Comment:
It seems like this line is not needed.

Reply:
Note that these are two different checks. The first checks to see if the `import_assertions` `v8::Local` is empty before we attempt to dereference the Local to check the length of the underlying Array. Basically, we only care about the check if the array, if provided, is not zero-length. We could combine these into a single statement tho